src/App.tsx

```diff
@@ -1,16 +1,109 @@
 import './App.scss';
+import { useEffect, useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
+import { IsMovie } from './types/Movie';
 import moviesFromServer from './api/movies.json';
 
+function areObjectPropertiesOnlySpaces(obj: {
+  [key: string]: string }): boolean {
+  return Object.keys(obj).some((key) => obj[key].trim().length === 0
+   && key !== 'description');
+}
+
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+  const [touchedMovies, setTouchedMovies] = useState<IsMovie>(
+    {
+      title: false,
+      imgUrl: false,
+      imdbUrl: false,
+      imdbId: false,
+    },
+  );
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+  const {
+    title,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = newMovie;
+
+  const onlySpaces = areObjectPropertiesOnlySpaces(newMovie);
+
+  const addOn = () => {
```

Comment:
```suggestion
  const handleMovieAdd = () => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364